src/cosmicqc/scdataframe.py

```diff
@@ -128,6 +136,280 @@ def export(
         else:
             raise ValueError("Unsupported file format for export.")
 
+    @staticmethod
+    def is_notebook_or_lab() -> bool:
+        """
+        Determines if the code is being executed in a Jupyter notebook or
+        JupyterLab environment.
+
+        This method attempts to detect the interactive shell environment
+        using IPython's `get_ipython` function. It checks the class name of the current
+          IPython shell to distinguish between different execution environments.
+
+        Returns:
+            bool:
+                - `True`
+                    if the code is being executed in a Jupyter notebook or JupyterLab.
+                - `False`
+                    otherwise (e.g., standard Python shell, terminal IPython shell,
+                    or scripts).
+        """
+        try:
+            # check for type of session via ipython
+            shell = get_ipython().__class__.__name__
+            if "ZMQInteractiveShell" in shell:
+                return True
+            elif "TerminalInteractiveShell" in shell:
+                return False
+            else:
+                return False
+        except NameError:
+            return False
+
+    def show_report(
+        self: SCDataFrame_type,
+        report_path: Optional[str] = None,
+        auto_open: bool = True,
+        color_palette: Optional[List[str]] = None,
+    ) -> None:
+        """
+        Generates and displays a report based on the current DataFrame's data
+        quality control (DQC) columns.
+
+        This method organizes the DQC columns from the DataFrame, creates
+        visualizations for each threshold set, and then either displays the
+        visualizations inline (if running in a Jupyter notebook or lab) or
+        opens an HTML report in the default web browser.
+
+        Args:
+            report_path (Optional[str]):
+                The file path where the HTML report should be saved and displayed.
+                If `None`, the report will be displayed inline if in a notebook
+                or lab environment.
```

Comment:
Love this!

Reply:
Cheers, thanks! It felt good to support both possibilities - the Jupyter environment is so powerful for analysis.